### ISSUE TITLE
[no-unused-vars] False positive on function parameter decorator with default value

### ISSUE THREAD

**Issue by LandazuriPaul** [typescript-eslint/typescript-eslint#2394]
#### Reproduction

I noticed that when the `@typescript-eslint/no-unused-vars)` rule is enabled (either `warn` or `error` mode) :

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": ["warn"]
  }
}
```

it triggers a **false positive** for function parameter decorators when the parameter for which it's used has a default value.

For example, in the following code case:

```TS
import { Controller, Get, Query } from '@nestjs/common';
import { AppService } from './app.service';

@Controller()
export class AppController {
  constructor(private readonly appService: AppService) {}

  @Get()
  getHello(@Query('foo') foo = 'bar'): string {
    console.log(foo);
    return this.appService.getHello();
  }
}
```

the `Query` decorator will be marked as unused (Line 1). But if we remove the default `'bar'` value, then there is no more false positive.

#### Expected Result
The linter shouldn't raise a warning/error on such circumstances.

#### Actual Result
The linter raises a warning/error.

#### Additional Info
I've created a [reproduction repository](https://github.com/LandazuriPaul/typescript-eslint-issue) with the above example.

#### Versions

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.9.0` |
| `@typescript-eslint/parser`        | `3.9.0` |
| `TypeScript`                       | `3.9.7` |
| `ESLint`                           | `7.6.0` |
| `node`                             | `12.18.3` |
| `npm`                              | `6.14.6` |

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/issues/1856

Please use the search.

**Comment by LandazuriPaul** [comment]
@bradzacher Thank you for pointing to the original issue.

I did « use the search », and didn't find it as it's quite meta and didn't appear for what I was looking for...
